src/network/transaction_request/mod.rs

```diff
@@ -18,75 +19,182 @@ pub struct TransactionRequest {
+        self.eip_712_meta
+            .get_or_insert_with(Eip712Meta::default)
+            .paymaster_params = Some(paymaster_params);
+    }
+
+    /// Builder-pattern method for setting paymaster params.
+    pub fn with_paymaster_params(mut self, paymaster_params: PaymasterParams) -> Self {
+        self.set_paymaster_params(paymaster_params);
+        self
+    }
+
+    /// Builder-pattern method for building a Zksync EIP-712 create2 tranasaction.
+    pub fn with_create2_params(
+        self,
+        salt: B256,
+        code: Vec<u8>,
+        constructor_data: Vec<u8>,
+        factory_deps: Vec<Vec<u8>>,
+    ) -> Result<Self, BytecodeHashError> {
+        self.with_create_params_inner(Some(salt), code, constructor_data, factory_deps)
+    }
+
+    /// Builder-pattern method for building a ZKsync EIP-712 create tranasaction.
+    pub fn with_create_params(
+        self,
+        code: Vec<u8>,
+        constructor_data: Vec<u8>,
+        factory_deps: Vec<Vec<u8>>,
+    ) -> Result<Self, BytecodeHashError> {
+        self.with_create_params_inner(None, code, constructor_data, factory_deps)
+    }
+
+    pub fn with_create_params_inner(
+        self,
+        salt: Option<B256>,
+        code: Vec<u8>,
+        constructor_data: Vec<u8>,
+        factory_deps: Vec<Vec<u8>>,
+    ) -> Result<Self, BytecodeHashError> {
+        let bytecode_hash = hash_bytecode(&code)?;
+        let factory_deps = factory_deps
+            .into_iter()
+            .chain(vec![code])
+            .map(Into::into)
+            .collect();
+        let input = crate::contracts::l2::contract_deployer::encode_create_calldata(
+            salt,
+            bytecode_hash.into(),
+            constructor_data.into(),
+        );
+        Ok(self
+            .with_to(CONTRACT_DEPLOYER_ADDRESS)
+            .with_input(input)
+            .with_factory_deps(factory_deps))
+    }
+}
+
+impl TransactionRequest {
+    #[deprecated(note = "use `set_paymaster_params` instead")]
+    pub fn set_paymaster(&mut self, paymaster_params: PaymasterParams) {
+        self.eip_712_meta
+            .get_or_insert_with(Eip712Meta::default)
+            .paymaster_params = Some(paymaster_params);
+    }
+
+    #[deprecated(note = "use `with_paymaster_params` instead")]
+    pub fn with_paymaster(mut self, paymaster_params: PaymasterParams) -> Self {
+        #[allow(deprecated)]
+        self.set_paymaster(paymaster_params);
+        self
+    }
+
+    #[deprecated(note = "use `with_create_params` instead")]
     pub fn zksync_deploy(
         self,
         code: Vec<u8>,
         constructor_data: Vec<u8>,
         factory_deps: Vec<Vec<u8>>,
     ) -> Result<Self, BytecodeHashError> {
+        #[allow(deprecated)]
         self.zksync_deploy_inner(None, code, constructor_data, factory_deps)
     }
 
+    #[deprecated(note = "use `with_create2_params` instead")]
     pub fn zksync_deploy_with_salt(
```

Comment:
I may be wrong, but AFAIR deployment with salt doesn't really follow `create2` semantics (e.g. it's still dependent on nonce). If that's the case, wouldn't the new name be misleading?

Reply:
create2 is purely dependent on `salt` so as to be deterministic. Can be verified on [system-contracts](https://github.com/matter-labs/era-contracts/blob/main/system-contracts/contracts/ContractDeployer.sol#L95) as well.

Reply:
But both methods seem to be invoking [`create`](https://github.com/popzxc/alloy-zksync/blob/main/src/contracts/l2/contract_deployer.rs#L12) method (which also accepts salt), no?

Reply:
The `salt` in create is purely for the signature as it's required by the bootloader. The [docs](https://github.com/matter-labs/era-contracts/blob/main/system-contracts/contracts/interfaces/IContractDeployer.sol#L65) mention that as well, in practice the nonce is used for address derivation in create, just like in EVM.

Reply:
> it still uses create but with salt :)

Aahhh I missed that we have a local copy of the interface, I think it makes sense to add a create2 here as well because calling create with `salt` is practically the same as calling it without salt.

Reply:
Sure, why not. Let's however do it in a separate PR.

Reply:
https://github.com/popzxc/alloy-zksync/pull/27